### PR TITLE
Fix explorer issue of not showing balance.

### PR DESCRIPTION
**Pull request by mikedoan**
1/ node syncing for explorer node.
2/ calculate block reward for explorer node.
3/ add setup for explorer node.

### CODE REVIEW

**Comment by LeoHChen** [comment]
Jenkins passed.  @mikedoan , Can you elaborate a bit more on this PR on what kind of issue resolved and why we need this PR?  It's hard to understand the logic behind the "more logic". Thanks.

**Comment by mikedoan** [comment]
Already did.

1/ node syncing for explorer node -> for explorer we do need to have state syncing but the explorer node does not need to send blocks for nodes like normal nodes(leader + validator)

2/ calculate block reward for explorer node. -> currently the blocks received from a explorer node do include reward info - we need to calculate the reward based on reward info attached to the blocks.

3/ add setup for explorer node.  -> fix some logic for the explorer service  /address end point.

File: api/service/explorer/service.go [hunk 1]
```diff
@@ -300,7 +289,16 @@ func (s *Service) GetExplorerAddress(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	result = data.Address
+	db := s.storage.GetDB()
+	bytes, err := db.Get([]byte(key))
+	if err != nil {
+		ctxerror.Warn(utils.GetLogger(), err, "cannot read address from db", "id", id)
```
Comment (LeoHChen) [inline api/service/explorer/service.go]: why different usage here? and line 299?

File: node/node_explorer.go [hunk 2]
```diff
@@ -81,6 +82,13 @@ func (node *Node) ExplorerMessageHandler(payload []byte) {
 	return
 }
 
+// AddNewBlockForExplorer add new block for explorer.
+func (node *Node) AddNewBlockForExplorer(block *types.Block) {
+	// TODO: Currently assumping blocks come in order.
```
Comment (LeoHChen) [inline node/node_explorer.go]: what's the impact if blocks didn't come in order?

File: api/service/explorer/service.go (hunk 1 above)
Comment (mikedoan) [reply api/service/explorer/service.go]: i kept the same logic of the original code. just move the logic of getbalance at the front.

File: node/node_explorer.go (hunk 2 above)
Comment (mikedoan) [reply node/node_explorer.go]: I'm writing another PR for this.
1/ i don't want to have a big PR
2/ i was telling another PR will come right after that - that's why i need to check in this one first.

**Pull request merged**